lib/src/typedef.dart

```diff
@@ -2,7 +2,7 @@ import 'dart:async';
 
 import 'package:flutter/widgets.dart';
 
-typedef FutureOr<Iterable<T>> SuggestionsCallback<T>(String pattern);
+typedef FutureOr<Iterable<T>> SuggestionsCallback<T>(String pattern, {int? page});
```

Comment:
having to write 
```dart
suggestionsCallback: (pattern, {page}) => /* ... */,
```
is very awkward.
for convenience I believe it would be beneficial to not make that parameter optional. 
```suggestion
typedef FutureOr<Iterable<T>> SuggestionsCallback<T>(String pattern, int? page);
```
that way, it can be ignored:
```dart
suggestionsCallback: (pattern, _) => /* ... */,
```

additionally, changing this callback I assume is a breaking change, so this can only be released with a new major version.

A better idea might be to add an alternative callback instead of changing this one.
The alternative callback can then be optional and we can add an assert to not specify both.

Reply:
@clragon , I think you are right. I'll add an alternative callback.